kmod/src/inode.c

```diff
@@ -1781,6 +1781,21 @@ static int try_delete_inode_items(struct super_block *sb, u64 ino)
 	return ret;
 }
 
+/*
+ * Keep calling try_ as long as we see that we had to back off after too
+ * many transactions under a single lock hold.
+ */
+static int keep_trying_delete_inode_items(struct super_block *sb, u64 ino)
+{
+	int ret;
+
+	do {
+		ret = try_delete_inode_items(sb, ino);
+	} while (ret == -EINPROGRESS);
```

Comment:
Are there any issues with this being a tight loop, and not something like delayed work?

Reply:
I don't think there's a problem here, though that's a good observation.  It's an unbroken loop to get all the work done.  The change here is to drop and re-acquire the cluster lock while doing that work.  The challenge is that as we re-acquire the lock we have to verify that it's still save to keep doing the work, that's what the body of try_ does before calling delete_inode_items to do the real item deletion work.